src/main/java/com/iexec/commons/poco/task/TaskDescription.java

```diff
@@ -153,6 +156,24 @@ public boolean requiresPreCompute() {
         return containsDataset() || containsInputFiles() || isBulkRequest();
     }
 
+    /**
+     * Returns if the SGX TEE framework is required
+     *
+     * @return {@literal true} if SGX is needed, {@literal false} otherwise
+     */
+    public boolean requiresSgx() {
+        return teeFramework == TeeFramework.SCONE || teeFramework == TeeFramework.GRAMINE;
+    }
+
+    /**
+     * Returns if the SGX TEE framework is required
```

Comment:
Small typo here I think, no ?
```suggestion
     * Returns if the TDX TEE framework is required
```